api/src/main/java/org/wildfly/transformer/ResourceTransformer.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2020 Red Hat, Inc, and individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.transformer;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * Resource transformer tries to convert given resource to another resource(s) by applying configured transformation rules.
+ * Can be used concurrently by multiple threads as instances of this class are thread safe.
```

Comment:
This comment about thread safety is pending whether Eclipse Transformer is thread safe and possibly when it might be updated to be thread safe.

Reply:
Eclipse integration doesn't implement (and there's no plan to) this ResourceTransformer contract @scottmarlow .